SparseGrids/Benchmarks/benchDifferentiate.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef _TASMANIAN_BENCHMARK_DIFFERENTIATE_HPP
+#define _TASMANIAN_BENCHMARK_DIFFERENTIATE_HPP
+
+#include "benchCommon.hpp"
+
+bool benchmark_differentiate(std::deque<std::string> &args){
+    if (args.size() < 9) return false;
+
+    // report the test parameters to reference later
+    cout << "differentiate";
+    for(auto &s : args) cout << " " << s;
+
+    auto grid_family = getGridFamily(args);
+    if (grid_family == GridFamily::none) return false;
+
+    int num_dimensions, num_outputs, num_depth, order, iteratons, num_jumps;
+    TypeDepth dtype;
+    TypeOneDRule rule;
+
+    auto riter = readEntries(args.begin(), num_dimensions, num_outputs, num_depth, dtype, rule, order, iteratons, num_jumps);
+
+    auto extra = extractWeightsLimits(grid_family, num_dimensions, dtype, riter, args.end());
+
+    auto make_grid = getLambdaMakeGrid(grid_family, num_dimensions, num_outputs, num_depth, dtype, rule, order, extra);
+
+    num_jumps = std::max(num_jumps, 1); // make at least one test
+
+    std::vector<std::vector<double>> inputs = getRandomVectors<double>(iteratons, num_dimensions, 1);
+
+    for(int jump = 0; jump < num_jumps; jump++){
+        auto grid = make_grid();
+        if (jump == 0) cout << " (points: " << grid.getNumPoints() << ")" << endl;
+        loadGenericModel(grid);
+        if (jump == 0) cout << "    note: reporting total time, does not normalize (divide) by iteratons\n";
```

Comment:
Make the time units here explicit, i.e.,
```
if (jump == 0) cout << "    note: reporting total time (ms), does not normalize (divide) by iteratons\n";
```

Reply:
All time is in ms in all cases. Added the comment.